### ISSUE TITLE
Unusable without first world internet

### ISSUE THREAD

**Issue by thcipriani** [buildabeer/buildabeer#21]
https://www.builda.beer/main.89d28366d85ecbba0c1b.bundle.js is 9.1MB after it loads (which takes 4 seconds for me) xhr requests for json models are triggered. The result is a 15 second load time for the new recipe screen even with relatively fast internet.

**Comment by Fencer04** [comment]
At what point do you see it that size? I'm only seeing it at 296B no matter what I'm doing.

**Comment by Fencer04** [comment]
@jkiv I see that it took 11seconds to load which is definitely slow but I'm seeing the size as 593K.

**Comment by jkiv** [comment]
@Fencer04  Oh, sorry, I was looking at just the total, didn't compare it to the file sizes. Something else is up then.

**Comment by buildabeer** [comment]
It'll probably be 2-3 weeks before I can look into this much, do you know what about main.js is causing it to bloat so much? Definitely not ideal.

**Comment by Fencer04** [comment]
Yes, I do see that now. That is really large but it seems to be cached for return visits. I'm not sure why it would be that large. I haven't taken a look at the code. I'm more of a Vue.js dev than Angular.

**Comment by zachhanson94** [comment]
Make sure AOT and build optimization are turned on during your build process. Had a similar issue recently until I realized I had disabled the feature. It should be on by default but it clearly isn’t working if your main js file is that large.

**Comment by buildabeer** [comment]
@zachhanson94 Awesome, glad it's been encountered before! I have a postinstall for heroku in package.json to do "ng build --aot --target=production", did I miss more that needed to happen?

**Comment by zachhanson94** [comment]
@buildabeer no I dont think you missed anything. After running tests I realized it was a lack of compression within the node server that was causing the file size to remain so high. I went ahead and already created a pull request which upgraded angular to current (7.1.1) and I am about to create another one that will include the compression as well as update the angular2-token module to its newest version which is now called angular-token. The angular-token required some changes throughout the codebase so many files were altered.
